src/Microsoft.DotNet.XHarness.Common/CLI/CommandArguments/Argument.cs

```diff
@@ -0,0 +1,275 @@
+            : base(prototype, description)
+        {
+        }
+
+        public override void Action(string argumentValue) => Value = argumentValue;
+    }
+
+    public abstract class RequiredStringArgument : Argument<string>
+    {
+        public RequiredStringArgument(string prototype, string description, string defaultValue = null!)
+            : base(prototype, description, defaultValue)
+        {
+        }
+
+        public override void Action(string argumentValue) => Value = argumentValue;
+
+        public override void Validate()
+        {
+            if (string.IsNullOrEmpty(Value))
+            {
+                throw new ArgumentException($"Required argument {Prototype} was not supplied");
+            }
+        }
+    }
+
+    public abstract class TimeSpanArgument : Argument<TimeSpan>
+    {
+        protected TimeSpanArgument(string prototype, string description, TimeSpan defaultValue)
+            : base(prototype, description)
+        {
+            Value = defaultValue;
+        }
+
+        public override void Action(string argumentValue)
+        {
+            if (int.TryParse(argumentValue, out var timeout))
+            {
+                Value = TimeSpan.FromSeconds(timeout);
+                return;
+            }
+
+            if (TimeSpan.TryParse(argumentValue, out var timespan))
+            {
+                Value = timespan;
+                return;
+            }
+
+            throw new ArgumentException($"{Prototype} must be an integer - a number of seconds, or a timespan (00:30:00)");
+        }
+    }
+
+    public abstract class PathArgument : StringArgument
+    {
+        private readonly bool _isRequired;
+
+        protected PathArgument(string prototype, string description, bool isRequired) : base(prototype, description)
+        {
+            _isRequired = isRequired;
+        }
+
+        public override void Action(string argumentValue) => Value = RootPath(argumentValue);
+
+        public override void Validate()
+        {
+            if (_isRequired && string.IsNullOrEmpty(Value))
+            {
+                throw new ArgumentException($"Required argument {Prototype} was not supplied");
+            }
+        }
+    }
+
+    public abstract class RequiredPathArgument : RequiredStringArgument
+    {
+        protected RequiredPathArgument(string prototype, string description) : base(prototype, description)
+        {
+        }
+
+        public override void Action(string argumentValue) => Value = RootPath(argumentValue);
+    }
+
+    public abstract class SwitchArgument : Argument<bool>
+    {
+        public SwitchArgument(string prototype, string description, bool defaultValue)
+            : base(prototype, description)
+        {
+            Value = defaultValue;
+        }
+
+        public override void Action(string argumentValue)
+        {
+            if (string.IsNullOrEmpty(argumentValue))
+            {
+                Value = !Value;
```

Comment:
what happens if I occasionally put such argument twice?

Reply:
Not sure what is the right behaviour. Maybe this would be better?
```suggestion
                Value = !defaultValue;
```

Reply:
I mean if I stored the value

Reply:
I think yes. Or somehow filter out more than one parameter because it's not obvious that the value would be unchanged after even amount of the repeated parameter